src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -0,0 +1,205 @@
+  }
+
+  private def validateWorkflowDescriptor(id: WorkflowId,
+                                         sourceFiles: WorkflowSourceFiles,
+                                         backend: Backend,
+                                         conf: Config): ErrorOr[WorkflowDescriptor] = {
+    val namespace = validateNamespace(sourceFiles.wdlSource, backend)
+    val rawInputs = validateRawInputs(id, sourceFiles.inputsJson)
+    val options = validateWorkflowOptions(id, sourceFiles.workflowOptionsJson, backend)
+    (namespace |@| rawInputs |@| options) { (_, _, _) } match {
+      case scalaz.Success((n, r, o)) =>
+        val validatedDescriptor = for {
+          c <- validateCoercedInputs(id, r, n).disjunction
+          d <- validateDeclarations(id, n, o, c, backend).disjunction
+        } yield WorkflowDescriptor(id, sourceFiles, o, r, n, c, d, backend, configCallCaching(conf), lookupDockerHash(conf))
+        validatedDescriptor.validation
+      case scalaz.Failure(f) => f.list.mkString("").failureNel
+    }
+  }
+
+  private def validateNamespace(source: WdlSource, backend: Backend): ErrorOr[NamespaceWithWorkflow] = {
+    try {
+      NamespaceWithWorkflow.load(source, backend.backendType).successNel
+    } catch {
+      case e: Exception => s"Workflow ${id.toString} unable to load namespace: ${e.getLocalizedMessage}".failureNel
+    }
+  }
+
+  private def validateWorkflowOptions(id: WorkflowId,
+                                      optionsJson: WorkflowOptionsJson,
+                                      backend: Backend): ErrorOr[WorkflowOptions] = {
+    WorkflowOptions.fromJsonString(optionsJson) match {
+      case Success(o) => buildValidateOptions(o, id, backend)
+      case Failure(e) => s"Workflow ${id.toString} contains bad options JSON: ${e.getLocalizedMessage}".failureNel
+    }
+  }
+
+  private def buildValidateOptions(options: WorkflowOptions, id: WorkflowId, backend: Backend): ErrorOr[WorkflowOptions] = {
+    val validatedOptions = options.successNel[String]
+    val backendOptionsDisjunction = for {
+      o <- validatedOptions.disjunction
+      b <- validateBackendOptions(id, o, backend).disjunction
+    } yield b
+    val backendOptions = backendOptionsDisjunction.validation
+    (validatedOptions |@| backendOptions) { (options, b) => options }
+  }
+
+  private def validateRawInputs(id: WorkflowId, json: WdlJson): ErrorOr[Map[String, JsValue]] = {
+    Try(json.parseJson) match {
+      case Success(JsObject(inputs)) => inputs.successNel
+      case _ => s"Workflow ${id.toString} contains bad inputs JSON: $json".failureNel
+    }
+  }
+
+  private def validateCoercedInputs(id: WorkflowId,
+                                    rawInputs: Map[String, JsValue],
+                                    namespace: NamespaceWithWorkflow): ErrorOr[WorkflowCoercedInputs] = {
+    namespace.coerceRawInputs(rawInputs) match {
+      case Success(r) => r.successNel
+      case Failure(e) => e.getLocalizedMessage.failureNel
```

Comment:
I honestly don't know why I used that, and it's throughout. Either something I started from here used it or something I cribbed off of at some point used it. 

I don't see the need to use it anywhere, do you?